src/ProjectNeon/Assets/Features/Effects/SetFeedUpEffect.cs

```diff
@@ -0,0 +1,44 @@
+﻿
+using System;
+
+/**
+ * Puts the decorated effect on queue and sets up a trigger so it is executed after
+ * the next card is played and card is of some specific type.
+ * 
+ * When some "feedType" card is played, this effect is triggered. If the "feedType" is
+ * different from this card, the effect is wasted.
+ * 
+ */
+public class SetFeedUpEffect : Effect
+{
+    private Effect _effect;
+
+    /**
+     * Feed type that triggers the effect execution
+     */
+    private FeedType _feedType;
+
+    public SetFeedUpEffect(Effect origin, FeedType feedType)
+    {
+        _effect = origin;
+        _feedType = feedType;
+    }
+
+    public SetFeedUpEffect(Effect origin, string attribute) : this(origin, (FeedType)Enum.Parse(typeof(FeedType), attribute))
+    {
+
+    }
+
+    public void Apply(Member source, Target target)
+    {
+        new QueueEffect(_effect).Apply(source, target);
+        BattleEvent.Subscribe<FeedCardResolutionStarted>(
```

Comment:
This one-shot idiom is still duplicated.

Reply:
@SilasReinagel Duplicated how? I'ts the only occurence in the code

Reply:
Okay. I can refactor it later. Just put a TODO in here. No puzzle needed. It's not a whole unit of work.